packages/adapter-knex/README.md

```diff
@@ -37,46 +15,61 @@ const keystone = new Keystone({
   name: 'My Awesome Project',
   adapter: new KnexAdapter(),
 });
-
-const uri = 'postgres://keystone5:k3yst0n3@127.0.0.1:5432/ks5_dev';
-const client = 'postgres';
-const schemaName = 'keystone';
-
-const knexOptions = { ... };
-
-keystone.connect(uri, {
-  client,
-  schemaName,
-  ...knexOptions,
-});
 ```
 
-## API
+## API `new KnexAdapter(options)`
 
-### `uri`
+### `schemaName`
 
-_**Default:**_ `'postgres://keystone5:k3yst0n3@127.0.0.1:5432/ks5_dev'`
+_**Default:**_ `'public'`
 
-Either a connection string, or a connection object, as accepted by Knex.
-See [knex docs](https://knexjs.org/#Installation-client) for more details.
-If the environment variable `KNEX_URI` is set, its value will be used as the default.
+All postgres tables are grouped within a schema and `public` is the default schema.
 
-### `client`
+### `knexOptions`
 
-_**Default:**_ `'postgres'`
+These options are passed directly through to Knex.
 
-Defines the type of backend to use. Current `postgres` is supported, but any value supported by Knex may be supported in the future.
+See the [knex docs](https://knexjs.org/#Installation-client) for more details.
 
-### `schemaName`
+_**Default:**_
 
-_**Default:**_ `'keystone'`
+```javaScript
+{
+  client: 'postgres',
+  connection: '<DEFAULT_CONNECTION_URL>'
+}
+```
 
-All keystone tables are grouped within a schema. This value should match the name of the schema used in the `CREATE SCHEMA` step above.
+The `DEFAULT_CONNECTION_URL` will be either one of the following environmental variables:
 
-### `knexOptions`
+- `CONNECT_TO`,
+- `DATABASE_URL`,
+- `KNEX_URI`
 
-Any extra options provided will be passed through to the Knex configuration function. See the [Knex docs](https://knexjs.org/#Installation-client) for possible values.
+or `'postgres://localhost/<DATABASE_NAME>'`where `DATABASE_NAME` is be derived from the Keystone project name.
 
 ## Debugging
 
 To log all Knex queries, run the server with the environment variable `DEBUG=knex:query`.
+
+## Setting Up a Database
+
+Before running Keystone with the Knex adapter you should configure a database. By default Knex will look for Postgres database on the default port, matching the project name, as the current user.
+
+In most cases this database will not exist and you will want configure a user and database for your application.
+
+Assuming you're on MacOS and have Postgres installed the `build-test-db.sh` does this for you:
+
+```sh
+./build-test-db.sh
+```
+
+Otherwise, you can run these steps manually:
+
+```shell
+createdb -U postgres keystone
+psql ks5_dev -U postgres -c "CREATE USER keystone5 PASSWORD 'k3yst0n3'"
+psql ks5_dev -U postgres -c "GRANT ALL ON DATABASE keystone TO keystone5;"
+```
+
+If using the above, you will want to set a connection string of: `postgres://keystone5:k3yst0n3@localhost:5432/keystone`
```

Comment:
This whole section feels like it'll be ultimately confusing for users; By following these instructions we're taking them away from the defaults we mentioned earlier. Now they have to know two different things just to get started?

Can I suggest we rename this section to something like `## Preparing for Production`:

```markdown
## Preparing for Production

When developing locally, using the defaults is fine. When it's time to deploy to production, your database may have a different name, and you will most likely require a username & password to access it.

Let's say your production database was setup with the following information:

- **Database host**: `awesome-pg-host.example.com`
- **Schema name**: `my-company`
- **Database name**: `keystone`
- **User**: `keystone5`
- **Pass**: `k3yst0n3` _(note: this is **not** a secure password!)_

Then to access your database, you would set the `schemaName` & `knexOptions`:

```javascript
const { KnexAdapter } = require('@keystonejs/adapter-knex');
const keystone = new Keystone({
  name: 'My Awesome Project',
  adapter: new KnexAdapter({
    schemaName: 'my-company',
    knexOptions: { connection: 'postgres://keystone5:k3yst0n3@awesome-pg-host.example.com/keystone' },
  }),
});
```

Then again, we can delete the `.sh` file, and remove any confusing language around running SQL commands, etc.

Reply:
Most user will want to setup up a database and I think having this with the knex adapter make some sense. The defaults may not work for everyone depending how they installed postgress and access on their machine.

I moved this to the bottom and I'd only expect people to be reading this if they need help.